### ISSUE TITLE
Performance decrease in recent builds (post 2.70rc1)

### ISSUE THREAD

**Issue by ETLegacyTeam** [etlegacy/etlegacy#188]
Author: @JanSimek (Radegast)
Date: 2013-02-01
Redmine Issue: 188, https://dev.etlegacy.com/issues/188

---

user:letter reported that he gets lower FPS on some maps like bremen than he did when he was using version:2.70rc1

We should test version:2.70rc1's performance and compare it to our recent builds.

---

Related commit(s):
c5c1522f6a851276cd4fea781fa55b936bfefbb3 - misc: rework of some compiler flags refs 193 refs..
 bde044c6bc2dcf1731c5db4be2301260210e63d6 - misc: do compiler flags for all OS refs 193 refs ..

**Comment by ETLegacyTeam** [comment]
From: @IR4T4 (IR4T4)
Date: 2013-02-02T19:23:05Z

---

letter, please report some more details. Did you just connect with ET: L to other online servers? ... used mod, how many fps are dropped? etc ... there are new features which might result in less fps so let's nail it down.

**Comment by ETLegacyTeam** [comment]
From: letter (letter)
Date: 2013-02-02T23:56:54Z

---

Well, I took some screenshots:
- attachment:1.jpg (I compiled ET:L with the latest files from repo)
- attachment:1.1.jpg and attachment:1.2.jpg (previous version but newer than RC1)

I spoke with Rafal and I asked him about his package of ET:L which I tested and I have to say It was much better but still FPS weren't as stable as in the previous version.

**Comment by ETLegacyTeam** [comment]
From: @IR4T4 (IR4T4)
Date: 2013-02-03T13:01:43Z

---

We shouldn't actually compare 1.jpg with 1.2 jpg ... the position and view direction of the spectator differ too much (see compass). There is more terrain on pic 1.2 and the tower on right side reduces VIS a lot (see lights from buildings) = more fps. If you watch the wall of 1st building in the background the wall is more detailed in 1 = less fps.

**Comment by ETLegacyTeam** [comment]
From: @rafal1137 (RaFaL)
Date: 2013-02-03T19:26:29Z

---

I think this is why we need better renderer even on W:ET was droping FPS no matter what kind of monstrous pc you had

**Comment by ETLegacyTeam** [comment]
From: Mateos (Mateos)
Date: 2013-02-03T19:28:48Z

---

You just need a good CPU, but for Vanilla only the GHz counts. I guess you've or will add parallelism support, maybe hyperthreading, and so it should be the best :)

I don't know if several rendering tasks can be switched to GPU?

**Comment by ETLegacyTeam** [comment]
From: letter (letter)
Date: 2013-02-03T22:37:41Z

---

IR4T4 wrote:
> We shouldn't actually compare 1.jpg with 1.2 jpg ... the position and view direction of the spectator differ too much (see compass). There is more terrain on pic 1.2 and the tower on right side reduces VIS a lot (see lights from buildings) = more fps. If you watch the wall of 1st building in the background the wall is more detailed in 1 = less fps.

It doesn't matter in which part of the map I was. Everywhere I had 70-90 FPS with the latest build of ET:L but with the previous one 125 FPS all the time. The best think is that I was using quake3.exe (renamed etl.exe) which gives me extra 20-30 FPS on every map. Now I use these files from Rafal and soon I will comapre the results.

Mateos wrote:
> You just need a good CPU, but for Vanilla only the GHz counts. I guess you've or will add parallelism support, maybe hyperthreading, and so it should be the best :)
> 
> I don't know if several rendering tasks can be switched to GPU?

Well, I've got i5 750 3.2GHz in turbo boost mode so I don't think I need a better CPU right now :)

**Comment by ETLegacyTeam** [comment]
From: @IR4T4 (IR4T4)
Date: 2013-02-04T11:22:11Z

---

I've found at least one reason why fps are 'dropped' with current code. Just after the 2.70rc1 release we did notice lightmap loading wasn't working properly. In other words: 2.70rc1 doesn't process map lightmaps. It's just the same as removing map/<MAPNAME>/lm_00XXX.tga files from a map (pk3) using lightmaps.
I did set the lighmap loading path to a bogus value in the current code and got a difference about +30 fps at my test position with same viewangel on bremen_final.

Test this: 

./etl +devmap bremen_final
r_lightmap 1

with rc1 and current code and you'll see the fixed bug
